includes/abstracts/abstract-wc-shipping-method.php

```diff
@@ -456,11 +462,13 @@ public function admin_options() {
 	public function get_option( $key, $empty_value = null ) {
 		// Instance options take priority over global options.
 		if ( $this->instance_id && array_key_exists( $key, $this->get_instance_form_fields() ) ) {
-			return $this->get_instance_option( $key, $empty_value );
+			$instance_option = apply_filters( 'woocommerce_shipping_' . $this->id . '_instance_option', $this->get_instance_option( $key, $empty_value ), $key, $empty_value, $this );
+			return $instance_option;
 		}
 
 		// Return global option.
-		return parent::get_option( $key, $empty_value );
+		$global_option = apply_filters( 'woocommerce_shipping_' . $this->id . '_global_option', parent::get_option( $key, $empty_value ), $key, $empty_value, $this );
+		return $global_option;
```

Comment:
To match my suggestion for instance option above, this should be named:

```
woocommerce_shipping_METHODID_option
```

and passed $key and $this. I don't see why empty value is relevant?

Reply:
Thank you for your reply, I'm working on those tweaks, I will push them later today. The $empty_value might be relevant, if you have some logic inside custom filter, that might sometimes need to know and return the empty value (for example if some condition is not met). It's not critical. Just let me know if it's a problem to have it there, so I could push the fix without / with the empty value inside the filter.

Reply:
If you were using this filter, you'd return the original value unless you actually want to change it. Thats why i question the need for the empty value.

Reply:
Ok, you have point. I will remove it 👍